lib/plugins/aws/package/compile/events/apiGateway/lib/requestValidator.js

```diff
@@ -0,0 +1,211 @@
+              Schema: definition,
+              ContentType: 'application/json',
+            },
+          },
+          [validatorLogicalId]: {
+            Type: 'AWS::ApiGateway::RequestValidator',
+            Properties: {
+              RestApiId: { Ref: this.apiGatewayRestApiLogicalId },
+              ValidateRequestBody: true,
+              ValidateRequestParameters: true,
+            },
+          },
+        });
+
+        if (modelName) {
+          this.serverless.service.provider.compiledCloudFormationTemplate.Resources[
+            modelLogicalId
+          ].Properties.Name = modelName;
+          this.serverless.service.provider.compiledCloudFormationTemplate.Resources[
+            validatorLogicalId
+          ].Properties.Name = validatorName;
+        }
+
+        if (description) {
+          this.serverless.service.provider.compiledCloudFormationTemplate.Resources[
+            modelLogicalId
+          ].Properties.Description = description;
+        }
+      }
+    }
+    this.validated.events.forEach(event => {
+      const resourceName = this.getResourceName(event.http.path);
+      const methodLogicalId = this.provider.naming.getMethodLogicalId(
+        resourceName,
+        event.http.method
+      );
+
+      // Old functionality
+      if (event.http.request && event.http.request.schema) {
+        const template = this.serverless.service.provider.compiledCloudFormationTemplate.Resources[
+          methodLogicalId
+        ];
+        for (const [contentType, schema] of _.entries(event.http.request.schema)) {
+          const modelLogicalId = this.provider.naming.getEndpointModelLogicalId(
+            resourceName,
+            event.http.method,
+            contentType
+          );
+
+          const validatorLogicalId = this.provider.naming.getEndpointValidatorLogicalId(
+            resourceName,
+            event.http.method
+          );
+
+          template.Properties.RequestValidatorId = { Ref: validatorLogicalId };
+          template.Properties.RequestModels = template.Properties.RequestModels || {};
+          template.Properties.RequestModels[contentType] = { Ref: modelLogicalId };
+
+          Object.assign(this.serverless.service.provider.compiledCloudFormationTemplate.Resources, {
+            [modelLogicalId]: {
+              Type: 'AWS::ApiGateway::Model',
+              Properties: {
+                RestApiId: this.provider.getApiGatewayRestApiId(),
+                ContentType: contentType,
+                Schema: schema,
+              },
+            },
+            [validatorLogicalId]: {
+              Type: 'AWS::ApiGateway::RequestValidator',
+              Properties: {
+                RestApiId: this.provider.getApiGatewayRestApiId(),
+                ValidateRequestBody: true,
+                ValidateRequestParameters: true,
+              },
+            },
+          });
+        }
+      }
+
+      // New Functionality
+      if (event.http.requestSchema) {
```

Comment:
Let's do:

```javascript
if (event.http.requestSchema) {
 ...
} else if (event.http.request && event.http.request.schema) {
 ...
}
```

At least I believe we should not support both formats put together, and if it's the case, new way should be favored